sql/sql_update.cc

```diff
@@ -740,183 +740,190 @@ int mysql_update(THD *thd,
       check_constant_expressions(values))
     read_removal= table->check_read_removal(select->quick->index);
 
-  while (!(error=info.read_record(&info)) && !thd->killed)
-  {
-    thd->inc_examined_row_count(1);
-    bool skip_record;
-    if (!select || (!select->skip_record(thd, &skip_record) && !skip_record))
+  error= table->file->ha_fast_update(thd, fields, values, conds);
+  if (error == 0)
+    error= -1; // error < 0 means really no error at all (see below)
+  else if (error != ENOTSUP) {
+    error= 1;
+  }
+  else
```

Comment:
Consider replacing "else" with a goto to jump over the legacy code to avoid its reindentation

Reply:
Using "goto" is not a good style unless it gives an advantage over other solutions. In our case I would prefer just not to do reindentation for "else" code block.

Reply:
It will not survive automated clang format checks. Goto would.

Reply:
There are a lot of places even in this function, which will not pass clang-format checks. The previous version was indented so that it would pass the check, but this is not good for our upstream merge process. The both solutions('goto' and not-indented 'else')  are bad from code readability perspective. But using 'goto' in the place where there is no need to use it only to pass clang-format check is quite strange solution from my point of view.